dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoFactory.java

```diff
@@ -128,9 +128,36 @@ public static SourceCodeRepoInterface createSourceCodeRepo(String gitUrl, String
      * @param url
      * @return a map with keys: Source, Username, Repository
      */
-    public static Map<String, String> parseGitUrl(String url) {
-        Matcher m1 = GITHUB_REGEX_PATTERN_1.matcher(url);
-        Matcher m2 = GITHUB_REGEX_PATTERN_2.matcher(url);
+    public static Optional<Map<String, String>> parseGitUrl(String url) {
+        return parseGitUrl(url, Optional.empty());
+    }
+
+    /**
+     * Parse Git URL to retrieve source, username and repository name.
+     *
+     * @param url
+     * @param repositoryName fixed repository name to match in git URL
+     * @return a map with keys: Source, Username, Repository
+     */
+    public static Optional<Map<String, String>> parseGitUrl(String url, Optional<String> repositoryName) {
+        String repositoryNameRegex1 = repositoryName.isPresent() ? repositoryName.get() : GITHUB_REPOSITORY_REGEX_1;
```

Comment:
Suggest moving GITHUB_REPOSITORY_REGEX_1 and GITHUB_REPOSITORY_REGEX_2 in here as a local constants. They're only used here in this method, and since you can no longer Pattern.compile in advance, it just makes the code easier to read.

Reply:
done